### PR TITLE
1.0.8 Stress / Longhaul Pipeline Fixes

### PR DESCRIPTION
**Pull request by and-rewsmith**
This PR accomplishes four things:

1. Fix errors in test configuration .yaml files
2. Consolidate all .yaml specifying pipeline steps into one branch so we won't have to point our tests at another branch's test configurations.
3. Backport some test functionality from master in order to fix broken windows e2e test script
4. Port the automated rocksDB mount that is about to go into 1.0.9

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# 1.0.8.2 (2019-09-23)
+## Edge Hub
+### Bug Fixes
+* Set RocksDB total WAL size to 512M ([07ee35d](https://github.com/Azure/iotedge/commit/07ee35d79a11eb6daab2a883e82ffbd7a1e93682))
+* Fix leaking event handlers in offline scenario ([f3b8a46](https://github.com/Azure/iotedge/commit/f3b8a463372d94163c286bd4c1a81ea9eb811832))
+* Fix error if client does not send product info  ([9267a05](https://github.com/Azure/iotedge/commit/9267a0545ac4394400afbcabdc123737ebcfbc17))
+
```
Comment (and-rewsmith) [inline CHANGELOG.md]: This block is already in 1.0.8 and appears due to a prior merge. See [here](https://github.com/Azure/iotedge/blob/release/1.0.8/CHANGELOG.md)

File: builds/e2e/templates/stresstest-deploy-windows.yaml [hunk 2]
```diff
@@ -39,6 +33,12 @@ steps:
       script: |
         #Requires -RunAsAdministrator
         $testName="Stress"
+        # schedule a task to stop iotedge service (terminate the test) in 4:30h
```
Comment (philipktlin) [inline builds/e2e/templates/stresstest-deploy-windows.yaml]: I recently updated to move schedule task block after run test script.  Please checkout the latest version from master.

File: builds/e2e/templates/stresstest-deploy-windows.yaml [hunk 3]
```diff
@@ -15,17 +15,11 @@ parameters:
...
   loadGen.message.frequency: ''
   loadGen1.transportType: 'Amqp_Tcp_Only'
-  loadGen2.transportType: 'Amqp_WebSocket_Only'
+  loadGen2.transportType: 'Amqp_Tcp_Only'
   loadGen3.transportType: 'Mqtt_Tcp_Only'
-  loadGen4.transportType: 'Mqtt_WebSocket_Only'
+  loadGen4.transportType: 'Mqtt_Tcp_Only'
 
 steps:
-  - task: CopyFiles@2
```
Comment (philipktlin) [inline builds/e2e/templates/stresstest-deploy-windows.yaml]: just add a simple comment here why remove "copy edgelet artifact"; same for other template files.

File: builds/e2e/templates/stresstest-deploy.yaml [hunk 4]
```diff
@@ -42,8 +36,8 @@ steps:
...
...tName "$testName" -containerRegistry "${{ parameters['container.registry'] }}" -containerRegistryUsername "${{ parameters['container.registry.username'] }}" -containerRegistryPassword "${{ parameters['container.registry.password'] }}" -iotHubConnectionString "${{ parameters['iotHub.connectionString'] }}" -eventHubConnectionString "${{ parameters['eventHub.connectionString'] }}" -snitchBuildNumber "${{ parameters['snitch.build.number'] }}" -snitchStorageAccount "${{ parameters['snitch.storage.account'] }}" -snitchStorageMasterKey "${{ parameters['snitch.storage.masterKey'] }}" -snitchAlertUrl "${{ parameters['snitch.alert.url'] }}" -loadGen1TransportType "${{ parameters['loadGen1.transportType'] }}" -loadGen2TransportType "${{ parameters['loadGen2.transportType'] }}" -loadGen3TransportType "${{ parameters['loadGen3.transportType'] }}" -loadGen4TransportType "${{ parameters['loadGen4.transportType'] }}" -amqpSettingsEnabled "${{ parameters['amqp.settings.enabled'] }}" -mqttSettingsEnabled "${{ parameters['mqtt.settings.enabled'] }}" -loadGenMessageFrequency "${{ parameters['loadGen.message.frequency'] }}" -snitchTestDurationInSecs "$(($(snitcher.test.duration.in.mins)*60))" -cleanAll
```
Comment (philipktlin) [inline builds/e2e/templates/stresstest-deploy.yaml]: get latest changes from master; i recently added a command to remove all existing job queued by at.

**Comment by philipktlin** [comment]
This will not fix DPS error since we didn't pick changes of IoTEdgeQuickstart and Leaf Device.  However Lorenzo said that 1.0.8.3 will be treated like LTS, it may be good to port IoTEdgeQuickstart and Leaf Device from master as well.  Let's discuss all together.

File: builds/e2e/templates/stresstest-deploy-windows.yaml (hunk 2 above)
Comment (and-rewsmith) [reply builds/e2e/templates/stresstest-deploy-windows.yaml]: I'm assuming this is only for our windows agents.

**Comment by and-rewsmith** [comment]
Philip brought up a good point: 

We should port all the deployment files from master before merging. Will reopen when done.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 5]
```diff
@@ -22,15 +22,28 @@ Set-Variable DockerServiceName -Value 'com.docker.service' -Option Constant
 
 Set-Variable EdgePackage -Value 'microsoft-azure-iotedge' -Option Constant
 
-Set-Variable EdgeInstallDirectory -Value "$env:ProgramFiles\iotedge" -Option Constant
+# If the user is running a 32-bit PS host on a 64-bit OS, then `$env:ProgramFiles` points to `C:\Program Files (x86)`
```
Comment (philipktlin) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Why do we need to update this file? is there any exception when call runE2ETests.sh?

Comment (and-rewsmith) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Not sure. I'll revert changes and see.

File: builds/e2e/longhaul.yaml [hunk 6]
```diff
@@ -202,14 +115,14 @@ jobs:
...
+            $(images.artifact.name.linux)/e2e_deployment_files/long_haul_deployment.template.json
             $(images.artifact.name.linux)/scripts/linux/runE2ETest.sh
             $(images.artifact.name.linux)/artifactInfo.txt
       # Deploy long haul
       - template: templates/longhaul-deploy.yaml
         parameters:
-          release.label: 'lh-mqtt'
-          edgelet.artifact.name: '$(edgelet.artifact.name)'
+          release.label: 'lh$(agent.group)'
+          edgelet.artifact.name: 'iotedged-debian9-arm32v7'
```
Comment (philipktlin) [inline builds/e2e/longhaul.yaml]:
```suggestion
          edgelet.artifact.name: '$(edgelet.artifact.name)'
```

Comment (philipktlin) [reply builds/e2e/longhaul.yaml]: you already updated the variable above.

File: e2e_deployment_files/long_haul_deployment.template.windows.json [hunk 7]
```diff
@@ -21,25 +21,19 @@
           "edgeAgent": {
             "type": "docker",
             "settings": {
-              "image": "<Container_Registry>/microsoft/azureiotedge-agent:<Build.BuildNumber>-linux-<Architecture>",
```
Comment (philipktlin) [inline e2e_deployment_files/long_haul_deployment.template.windows.json]: This looks weird to me; why windows deployment file uses linux image.

File: edgelet/contrib/config/windows/config.yaml [hunk 8]
```diff
@@ -111,7 +111,7 @@ agent:
   type: "docker"
   env: {}
   config:
-    image: "mcr.microsoft.com/azureiotedge-agent:1.0.8.1"
+    image: "mcr.microsoft.com/azureiotedge-agent:1.0.8.2"
```
Comment (philipktlin) [inline edgelet/contrib/config/windows/config.yaml]: why have this change?

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 9]
```diff
@@ -1944,4 +1944,4 @@ Export-ModuleMember `
     -Alias `
         Install-SecurityDaemon,
         Uninstall-SecurityDaemon
-}
+}
```
Comment (philipktlin) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: revert this change

File: versionInfo.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "1.0.8.1",
+  "version": "1.0.8.2",
```
Comment (philipktlin) [inline versionInfo.json]: why there is a difference? i think your branch is from release\1.0.8, did you merge latest changes from release\1.0.8 to your PR branch?

File: edgelet/contrib/config/windows/config.yaml (hunk 8 above)
Comment (and-rewsmith) [reply edgelet/contrib/config/windows/config.yaml]: This is an error in the diff. This change is already present in 1.0.8. I'm not sure why it is showing here.

File: e2e_deployment_files/long_haul_deployment.template.windows.json (hunk 7 above)
Comment (and-rewsmith) [reply e2e_deployment_files/long_haul_deployment.template.windows.json]: Whatever was happening here was standard practice for 1.0.8. If I had to guess, at the time of the first 1.0.8 release windows images were not available and no one has ported an update until now.

File: versionInfo.json (hunk 10 above)
Comment (and-rewsmith) [reply versionInfo.json]: This is an error in the diff. Not sure why it's happening. I talked to Damon and he says it sometimes occurs when you merge commits from a branch, then open a PR into that same branch.

File: builds/e2e/longhaul.yaml (hunk 6 above)
Comment (and-rewsmith) [reply builds/e2e/longhaul.yaml]: Good catch!

**Comment by philipktlin** [comment]
As the test resulof deployment (running e2e test script) from vsts is positive, please approve it and merge to release/1.0.8
